app/components/chat/BaseChat.tsx

```diff
@@ -200,29 +207,24 @@ export const BaseChat = React.forwardRef<HTMLDivElement, BaseChatProps>(
       setApiKeys(newApiKeys);
       Cookies.set('apiKeys', JSON.stringify(newApiKeys));
 
-      const provider = LLMManager.getInstance(import.meta.env || process.env || {}).getProvider(providerName);
+      setIsModelLoading(providerName);
 
-      if (provider && provider.getDynamicModels) {
-        setIsModelLoading(providerName);
+      try {
+        const providerSettings = getProviderSettings();
 
-        try {
-          const providerSettings = getProviderSettings();
-          const staticModels = provider.staticModels;
-          const dynamicModels = await provider.getDynamicModels(
-            newApiKeys,
-            providerSettings,
-            import.meta.env || process.env || {},
-          );
-
-          setModelList((preModels) => {
-            const filteredOutPreModels = preModels.filter((x) => x.provider !== providerName);
-            return [...filteredOutPreModels, ...staticModels, ...dynamicModels];
-          });
-        } catch (error) {
-          console.error('Error loading dynamic models:', error);
-        }
-        setIsModelLoading(undefined);
+        const response = await fetch('/api/models', {
+          headers: {
+            'x-client-api-keys': JSON.stringify(newApiKeys),
+            'x-client-provider-settings': JSON.stringify(providerSettings),
+          },
+        });
+        const data = await response.json();
+        const typedData = data as { modelList: ModelInfo[] };
+        setModelList(typedData.modelList);
+      } catch (error) {
+        console.error('Error loading dynamic models:', error);
       }
```

Comment:
for this one, I think we better provide the provider name to indicate the endpoint to only provide models list for that particular provider, this should stop us loading all the models again which did not had any changes

Reply:
I enhanced the log and provided a route to only get the specific model data